app/services/efile/gyr_efiler_service.rb

```diff
@@ -1,44 +1,36 @@
 
     def self.run_efiler_command(*args)
-      raise StandardError.new("Cannot be used from the test environment") if Rails.env.test?
-
-      Dir.mktmpdir do |working_directory|
-        FileUtils.mkdir_p(File.join(working_directory, "output", "log"))
-        ensure_config_dir_prepared
-
-        # TODO: If the process blocks for >10 min, terminate it.
-        # TODO: Send process stdout to logs.
-        # TODO: Send output/logs/ to logs after process terminates.
-        classes_zip_path = Dir.glob(Rails.root.join("vendor", "gyr_efiler", "gyr-efiler-classes-#{CURRENT_VERSION}.zip"))[0]
-        raise StandardError.new("You must run rails setup:download_gyr_efiler") if classes_zip_path.nil?
+      ensure_efiler_has_been_downloaded
+      ensure_java_is_installed
+      ensure_config_dir_is_prepared
 
-        config_dir = Rails.root.join("tmp", "gyr_efiler", "gyr_efiler_config").to_s
+      # Since gyr-efiler saves log output to output/log/* within its current working directory, always launch it from
+      # a new temporary working directory. We cannot chdir() within here because the app has other
+      # threads that would be affected by chdir() so file accesses must be prefixed by this directory.
+      Dir.mktmpdir do |efiler_working_directory|
+        FileUtils.mkdir_p(File.join(efiler_working_directory, "output", "log"))
 
-        # On macOS, "java" will show a confusing pop-up if you run it without a JVM installed. Check for that and exit early.
-        if File.exists?("/Library/Java/JavaVirtualMachines") && Dir.glob("/Library/Java/JavaVirtualMachines/*").empty?
-          raise StandardError.new("Seems you are on a mac & lack Java. Run: brew tap AdoptOpenJDK/openjdk && brew install adoptopenjdk8")
-        end
-        # /Library/Java/JavaVirtualMachines
-        java = ENV["VITA_MIN_JAVA_HOME"] ? File.join(ENV["VITA_MIN_JAVA_HOME"], "bin", "java") : "java"
-
-        argv = [java, "-cp", classes_zip_path, "org.codeforamerica.gyr.efiler.App", config_dir, *args]
+        argv = [java_path, "-cp", classes_zip_path, "org.codeforamerica.gyr.efiler.App", gyr_efiler_config_dir, *args]
         puts "Running: #{argv.inspect}"
         pid = Process.spawn(*argv,
           unsetenv_others: true,
-          chdir: working_directory,
+          chdir: efiler_working_directory,
           in: "/dev/null"
         )
         Process.wait(pid)
         raise StandardError.new("Process failed to exit?") unless $?.exited?
 
         exit_code = $?.exitstatus
-        raise StandardError.new("gyr-efiler failed; exited with #{exit_code}") if exit_code != 0
+        if exit_code == 0
+          return get_single_file_from_zip(Dir.glob(File.join(efiler_working_directory, "output", "*.zip"))[0])
+        end
 
-        get_single_file_from_zip(Dir.glob(File.join(working_directory, "output", "*.zip"))[0])
+        raise StandardError.new(File.read(File.join(efiler_working_directory, 'output/audit_log.txt')))
```

Comment:
In the file that saves this into raw response, it might be nice to output only the message instead of the whole error object like i (think) we're doing at the moment (in another file). 

However, I bias towards a release so that we go into the weekend with this issue resolved than making that change now.